File: Interfaces/ISrtImportLineService.cs
using WMS_WEBAPI.DTOs;

namespace WMS_WEBAPI.Interfaces
{
    public interface ISrtImportLineService
    {
        Task<ApiResponse<IEnumerable<SrtImportLineDto>>> GetAllAsync();
        Task<ApiResponse<SrtImportLineDto>> GetByIdAsync(long id);
        Task<ApiResponse<IEnumerable<SrtImportLineDto>>> GetByHeaderIdAsync(long headerId);
        Task<ApiResponse<IEnumerable<SrtImportLineDto>>> GetByLineIdAsync(long lineId);
        Task<ApiResponse<IEnumerable<SrtImportLineDto>>> GetByRouteIdAsync(long routeId);
        Task<ApiResponse<IEnumerable<SrtImportLineDto>>> GetByStockCodeAsync(string stockCode);
        Task<ApiResponse<IEnumerable<SrtImportLineDto>>> GetByErpOrderNoAsync(string erpOrderNo);
        Task<ApiResponse<SrtImportLineDto>> CreateAsync(CreateSrtImportLineDto createDto);
        Task<ApiResponse<SrtImportLineDto>> UpdateAsync(long id, UpdateSrtImportLineDto updateDto);
        Task<ApiResponse<bool>> SoftDeleteAsync(long id);
    }
}

File: WebApi/Interfaces/SubcontractingReceiptTransferInterfaces/ISrtImportLineService.ts
import type { CreateSrtImportLineDto, SrtImportLineDto, UpdateSrtImportLineDto } from '../../Models/index';
import type { ApiResponse, PagedResponse } from '../../Models/ApiResponse';

export interface ISrtImportLineService {
  getAll(): Promise<ApiResponse<SrtImportLineDto[]>>;
  getById(id: number): Promise<ApiResponse<SrtImportLineDto>>;
  getByHeaderId(headerId: number): Promise<ApiResponse<SrtImportLineDto[]>>;
  getByLineId(lineId: number): Promise<ApiResponse<SrtImportLineDto[]>>;
  getByStockCode(stockCode: string): Promise<ApiResponse<SrtImportLineDto[]>>;
  create(createDto: CreateSrtImportLineDto): Promise<ApiResponse<SrtImportLineDto>>;
  update(id: number, updateDto: UpdateSrtImportLineDto): Promise<ApiResponse<SrtImportLineDto>>;
  softDelete(id: number): Promise<ApiResponse<boolean>>;
}
